### PR TITLE
feat: Adding E2E metric test for RDS database interactions

### PR DESCRIPTION
**Pull request by jerry-shao**
*Issue description:*

The E2E test workflow now calls `/mysql` API which generates a SQL query to RDS database. However, we are not testing the telemetries ingested by this Agent yet.

*Description of changes:*

This is the first change to test telemetry ingested by Application Signals agent for SQL query interruptions.

This change calls CloudWatch Metrics for ListMetrics API and compares with the expected metric list.

E2E test workflow for us-east-1: https://github.com/jerry-shao/aws-application-signals-test-framework/actions/runs/9961505823/job/27523164038

No failure:
image

---
*Ensure you've run the following tests on your changes and include the link below:*
To do so, create a `test.yml` file with `name: Test` and workflow description to test your changes, then remove the file for your PR. Link your test run in your PR description. This process is a short term solution while we work on creating a staging environment for testing.

NOTE: TESTS RUNNING ON A SINGLE EKS CLUSTER CANNOT BE RUN IN PARALLEL. See the [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) keyword to run tests in succession.
- Run Java EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run Python EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run metric limiter on EKS cluster `e2e-playground` in us-east-1 and eu-central-2
- Run EC2 tests in all regions
- Run K8s on a separate K8s cluster (check IAD test account for master node endpoints; these will change as we create and destroy clusters for OS patching)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by majanjua-amzn — COMMENTED** [review]
1. Why no log or trace validation?
2. Can we see the test in more than one region?
3. Some proof that the resources in the accounts that run the canary are configured in a way that will accept the same changes as these that make the test pass in your dev environment would also be great. You can use the testing instructions in the automated PR description for that, although you're on a fork so it won't work...

**Comment by jerry-shao** [comment]
@majanjua-amzn Thanks for your review!

> Why no log or trace validation?
log and trace validation will be in separate PRs. They are in draft mode at the moment, I will polish them once this one is merged.

> Can we see the test in more than one region?
Em... looks like it's hard to have infra in two regions together. I may need to delete my us-east-1 stack and re-deploy to another region and modify the `region-account` secrets.

> Some proof that the resources in the accounts that run the canary are configured in a way that will accept the same changes as these that make the test pass in your dev environment would also be great. You can use the testing instructions in the automated PR description for that, although you're on a fork so it won't work...
Yep, the testing infra mentioned in the PR description is hard for me.

The infra created in my personal account is using the CDK package to deploy a personal stack. Then I created a secret stored in secrets manager called `region-account/us-east-1` and the credential is basically my dev-account so it will use my account as the ACCOUNT_ID env variable through all workflow steps. Nothing else special from my infra.

**Comment by jerry-shao** [comment]
Change has been running continuously every 15 minutes for two days: https://github.com/jerry-shao/aws-application-signals-test-framework/actions?query=branch%3Amain

The missing runs were due to a known problem: docker throttling
E.g: https://github.com/jerry-shao/aws-application-signals-test-framework/actions/runs/10033641691/job/27727046487

CloudWatch metric:

image

File: .github/workflows/java-eks-e2e-test.yml [hunk 1]
```diff
@@ -374,6 +374,7 @@ jobs:
           --service-name sample-application-${{ env.TESTING_ID }}
           --remote-service-name sample-remote-application-${{ env.TESTING_ID }}
           --remote-service-deployment-name ${{ env.REMOTE_SERVICE_DEPLOYMENT_NAME }}
+          --remote-resource-identifier "information_schema|${{env.RDS_MYSQL_CLUSTER_ENDPOINT}}|3306"
```
Comment (majanjua-amzn) [inline .github/workflows/java-eks-e2e-test.yml:377]: Just confirming, have we double checked that this value is consistently this shape? What is the significance of `3306` here, is it always the same number?

**Review by majanjua-amzn — APPROVED** [review]

File: .github/workflows/java-eks-e2e-test.yml (hunk 1 above)
Comment (jerry-shao) [reply .github/workflows/java-eks-e2e-test.yml:377]: Port is always 3306 in the RDS cluster we created.

**Pull request merged**